### ISSUE TITLE
artifacts do not clean .cache after upload

### ISSUE THREAD

**Issue by maria-korosteleva** [wandb/wandb#1193]
`wandb --version && python --version && uname`

* Weights and Biases version: 0.9.4
* Python version: 3.6.10
* Operating System: Win 10

### Description

Artifacts upload in wandb runs doesn't seem to clean up cache files after itself properly. 

As I work with my ML experiments, the `C:\Users\<Username>\.cache\wandb\artifacts`  folder grows up to the point that it exhausts all the free space on my drive (up to 50 Gb). After this point, I have to clean it manually to resume my work. 

The size of all artifacts to be uploaded within a single run is about 5Gb (counting all the versions of the same files), which is definitely less that the .cache folder size when I discover the problem

### Additional errors

At times I get these errors at the end of the run: 

```
C:\ProgramData\Anaconda3\lib\site-packages\wandb\compat\tempfile.py:64: UserWarning: Couldn't remove temp directory C:\Users\Maria\AppData\Local\Temp\tmp7apg1zqmwandb
  _warnings.warn("Couldn't remove temp directory %s" % name)
C:\ProgramData\Anaconda3\lib\site-packages\wandb\compat\tempfile.py:64: UserWarning: Couldn't remove temp directory C:\Users\Maria\AppData\Local\Temp\tmpxa113_1ewandb-media
  _warnings.warn("Couldn't remove temp directory %s" % name)
C:\ProgramData\Anaconda3\lib\site-packages\wandb\compat\tempfile.py:64: UserWarning: Couldn't remove temp directory C:\Users\Maria\AppData\Local\Temp\tmpghqvixfewandb
  _warnings.warn("Couldn't remove temp directory %s" % name)
c:\programdata\anaconda3\lib\site-packages\wandb\compat\tempfile.py:64: UserWarning: Couldn't remove temp directory C:\Users\Maria\AppData\Local\Temp\tmpx0vikdffwandb-media
  _warnings.warn("Couldn't remove temp directory %s" % name)
c:\programdata\anaconda3\lib\site-packages\wandb\compat\tempfile.py:64: UserWarning: Couldn't remove temp directory C:\Users\Maria\AppData\Local\Temp\tmpit2fs8f5wandb
  _warnings.warn("Couldn't remove temp directory %s" % name)
```

Which seems related to cleaning-up but point to a different Temp folder that DO NOT have this problem of size growing

**Comment by tyomhak** [comment]
Thanks for reporting! @annirudh  can you please take a look?

**Comment by shawnlewis** [comment]
You're right, the artifacts cache grows forever right now. You can feel free to remove the folder at any time. Artifacts will redownload files it needs.

We'll work on smarter cache management strategies over the next few months.

**Comment by maria-korosteleva** [comment]
@shawnlewis thank you for the workaround tip!

**Comment by maria-korosteleva** [comment]
`.cache` is still never cleaned up,  wandb v.0.10.12

**Comment by clarkbab** [comment]
Hi @maria-korosteleva,

I had the same problem and was able to change the cache directory to another drive available to me. If that's an option for you, look into setting the environment variable `WANDB_CACHE_DIR` (see code [here](https://github.com/wandb/client/blob/d1286556836a25d68f21816e5340596515af773a/wandb/env.py#L257)). 
The env var name has changed to  `CACHE_DIR` in recent versions of `wandb`.

**Comment by turian** [comment]
@sydholl I still have this issue, just curious why it got closed?

**Comment by spirosbax** [comment]
I also still have the issue. Artifacts get stored in `.cache/wandb` and don't get automatically deleted. When multiple large models for hundreds of epochs this results in >200GBs of artifacts. For now I run `wandb artifact cache cleanup 10GB` a couple of times per day. Is there some other way to fix this?

**Comment by jasonkhadka** [comment]
@sydholl Can you please explain why is this closed? This problem still persists. 
```
wandb, version 0.13.10
Python 3.8.16
Linux
```
A new feature to limit the cache size will be highly appreciated.

**Comment by turian** [comment]
@shawnlewis Wondering if it would be possible to reopen?

**Comment by turian** [comment]
@Twilightdonkey those docs give little guidance about whether it is safe to do this during a long-running Lightning job that has wandb logging. GPT4 suggests that this might not be a safe operation if logging is still happening. Nor does the code give any guidance about whether this is a safe operation:

https://github.com/wandb/wandb/blob/a4aa143837e7a43d1f3b11fa3100b1f8e63d27f3/wandb/sdk/interface/artifacts.py#L936

**Comment by turian** [comment]
And in fact, my very issue is that long running jobs exhaust disk.

**Comment by n-balla** [comment]
I have the same issue. Long-running jobs exhaust disk! I hope the .cache files get deleted automatically!

**Comment by benrhodes26** [comment]
This is still a significant issue.

**Comment by Twilightdonkey** [comment]
Note that as for wandb version 0.19 the artifact and sdk interface has changed
```python
from wandb.sdk.artifacts.artifact_file_cache import get_artifact_file_cache

target_size = int(10e9) # Removes cache size to 10 GB
cache = get_artifact_file_cache()
cache.cleanup(target_size)
```

**Comment by alex-schutz** [comment]
This continues to be an issue
